File: old/rin.Framework/Graphics/Windows/Events/CursorButtonEvent.cs
namespace rin.Framework.Graphics.Windows.Events;

public class CursorButtonEvent : CursorEvent
{
    public required CursorButton Button;
    public required InputState State;
    public required InputModifier Modifiers;
    public bool IsAltDown => Modifiers.HasFlag(InputModifier.Alt);
    public bool IsControlDown => Modifiers.HasFlag(InputModifier.Control);
    public bool IsShiftDown => Modifiers.HasFlag(InputModifier.Shift);
}

File: lib/rin/gui/events/CursorButtonEvent.cpp
#include "rin/gui/events/CursorButtonEvent.h"
namespace rin::gui
{
    CursorButtonEvent::CursorButtonEvent(Surface* inSurface, const io::CursorButton& inButton, const io::InputState& inState, const Vec2<>& inPosition,
        const Flags<io::InputModifier>& inModifiers)   : Event(inSurface), button(inButton), state(inState), position(inPosition), modifiers(inModifiers)
    {
    }
}
